extensions/default/src/DicomWebDataSource/index.js

```diff
@@ -24,123 +22,136 @@ const { naturalizeDataset, denaturalizeDataset } = DicomMetaDictionary;
-          dicomWebConfig.acceptHeader,
-          dicomWebConfig.requestTransferSyntaxUID,
-          dicomWebConfig.omitQuotationForMultipartRequest
-        );
-
-        return {
-          ...authorizationHeader,
-          Accept: formattedAcceptHeader,
-        };
-      };
-
-      qidoConfig = {
-        url: dicomWebConfig.qidoRoot,
-        staticWado: dicomWebConfig.staticWado,
-        singlepart: dicomWebConfig.singlepart,
-        headers: userAuthenticationService.getAuthorizationHeader(),
-        errorInterceptor: errorHandler.getHTTPErrorHandler(),
-      };
-
-      wadoConfig = {
-        url: dicomWebConfig.wadoRoot,
-        staticWado: dicomWebConfig.staticWado,
-        singlepart: dicomWebConfig.singlepart,
-        headers: userAuthenticationService.getAuthorizationHeader(),
-        errorInterceptor: errorHandler.getHTTPErrorHandler(),
-      };
-
-      // TODO -> Two clients sucks, but its better than 1000.
-      // TODO -> We'll need to merge auth later.
-      qidoDicomWebClient = dicomWebConfig.staticWado
-        ? new StaticWadoClient(qidoConfig)
-        : new api.DICOMwebClient(qidoConfig);
-
-      wadoDicomWebClient = dicomWebConfig.staticWado
-        ? new StaticWadoClient(wadoConfig)
-        : new api.DICOMwebClient(wadoConfig);
+      },
     },
     query: {
       studies: {
         mapParams: mapParams.bind(),
         search: async function (origParams) {
-          qidoDicomWebClient.headers = getAuthrorizationHeader();
-          const { studyInstanceUid, seriesInstanceUid, ...mappedParams } =
-            mapParams(origParams, {
-              supportsFuzzyMatching: dicomWebConfig.supportsFuzzyMatching,
-              supportsWildcard: dicomWebConfig.supportsWildcard,
-            }) || {};
-
-          const results = await qidoSearch(qidoDicomWebClient, undefined, undefined, mappedParams);
-
+          const clients = clientManager.getClientsForQidoRequests();
+          // concatenate series metadata from all servers
+          const clientResultsPromises = clients.map(client => {
+            const { studyInstanceUid, seriesInstanceUid, ...mappedParams } =
+              mapParams(origParams, {
+                supportsFuzzyMatching: client.supportsFuzzyMatching,
+                supportsWildcard: client.supportsWildcard,
+              }) || {};
+            let clientResults;
+            try {
+              clientResults = qidoSearch(
+                client.qidoDicomWebClient,
+                undefined,
+                undefined,
+                mappedParams
+              );
+            } catch {
+              clientResults = [];
+            }
+            return clientResults;
+          });
+          await Promise.allSettled(clientResultsPromises);
+
+          let results = [];
+          const studyInstanceUIDs = {};
+          // here remove duplicate studies
+          for (let i = 0; i < clientResultsPromises.length; i++) {
```

Comment:
A nested loop feels a little out of place here. Consider doing a map followed by a reduce OR a forEach

```
const clientResults = clientResultsPromises.map(resultPromise=> await resultPromise);
clientResults.forEach((clientResult) => {
  const studyInstanceUID = clientResult[STUDY_INSTANCE_UID].Value[0];
  if (!(studyInstanceUID in studyInstanceUIDs)) {
    studyInstanceUIDs[studyInstanceUID] = clientResult;
    results.push(clientResult);
  }
})
```